tools/scripts/check_test_lst_paths.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# This script checks tests/std/test.lst (and tr1) for duplicate, missing, nonexistent, and unsorted test paths.
+
+from pathlib import Path
+import sys
+
+def is_nonempty_dir(p: Path) -> bool:
+    return p.is_dir() and any(p.iterdir())
+
+if __name__ == "__main__":
+    if len(sys.argv) != 2:
+        sys.exit(f"Usage: python {sys.argv[0]} tests/std-OR-tr1")
+
+    absolute_base = Path(sys.argv[1]).absolute()

```

Comment:
Not blocking: I think this won't normalize paths. It probably doesn't matter here, but we could use `resolve` instead?